src/main/java/org/jabref/logic/pdf/search/indexing/DocumentReader.java

```diff
@@ -122,6 +125,20 @@ private void addContentIfNotEmpty(PDDocument pdfDocument, Document newDocument)
             if (StringUtil.isNotBlank(pdfContent)) {
                 newDocument.add(new TextField(CONTENT, pdfContent, Field.Store.YES));
             }
+            pdfDocument.getPages().forEach(page -> {
+                try {
+                    for (PDAnnotation annotation : page.getAnnotations( annotation -> {
+                        if (annotation.getContents() == null) {
+                            return false;
+                        }
+                        return annotation.getSubtype().equals("Text") || annotation.getSubtype().equals("Highlight");
+                    })) {
+                        newDocument.add(new TextField(ANNOTATIONS, annotation.getContents(), Field.Store.YES));
+                    }
+                } catch (IOException e) {
+                    e.printStackTrace();
```

Comment:
LOGGER

Reply:
When will I learn... sorry.
Instead of logging the exception here I let it fall through to the other catch-block. Since it acesses the same file, the IO should happen on the earlier call anyways (at pdfTextStripper.getText).